public/characterCreator.html

```diff
@@ -1 +1,157 @@
+                }
+                return result; // returns statpoint cost in the case of it being a non negative.
+            }
+        }
+        /** Returns the net cost of set statlevels with current ammount of statpoitns, returns negative if their is a deficit and positive if there is a suprlus of statpoints
+         * @param {htmlcollection} statCollection This is a collection of elements that contain number values within the class 'stats'
+         * @return {number} workingValue This is a number that is the total statpoint cost being either negative or positive depending on if free statpoints remain.
+         */
+        function calculateDeficit(statCollection) {
+            let workingValue = 0;
+            for (statValue of statCollection) {
+                if (isNaN(statValue.value)) {
+                    return "Invalid entry";
+                }
+                workingValue += generateStatCost(statValue.value);
+            }
+            const setStatPoints = parseInt(document.getElementById('statpoints').value);
+            workingValue = setStatPoints - workingValue;
+            return workingValue;
+        }
+
+        /** Listens to the form submission, takes the values of the form, puts them into json, and puts it into a cookie called "value".
+         * @param {event} event The submission of the 'characterCreator' form, captured and instead of being sent to an endpoint, it is set as the value of the cookie "value".
+         */
+        document.getElementById('characterCreator').onsubmit = function(event) {
+            event.preventDefault();
+            const elements = document.getElementsByClassName("stats");
+            if (calculateDeficit(elements) < 0) {
+                alert("Not enough stat points for selected stats.  Check your remaining points.");
+            } else {
+
+                const data = new FormData(event.target);
+                const value = JSON.stringify(Object.fromEntries(data.entries()));
+
+                //Put this JSON string into a cookie
+                setCookie("value", value, 365);
+
+                //and take us to the character sheet page to see our new character stats
+                window.location.href = "characterSheet.html";
+            }
+        };
+
+        /** Takes in the cookie name, value, and number of days to keep it before deletion.  Creates on the local path=/
+         * @param {string} cname This is the name of the cookie, which will be used to reference it when accessing it
+         * @param {string} cvalue The string that will be set as the cookie's value, in this specific usage, a stringified JSON object containing the character's information taken from the HTML form
+         * @param {integer} exdays Number of days the browser should store this cookie.  Broken down from days into milliseconds
+         */
+        function setCookie(cname, cvalue, exdays) {
+            var d = new Date();
```

Comment:
var

Reply:
d

Reply:
no comments?

Reply:
made it all inline, and commented on the static multiplication